### PR TITLE
feat: add error message to breakdown/cli

### PR DESCRIPTION
**Pull request by Pauluson**
#535 
`<pre>make -s run ARGS=&quot;breakdown --path plan&quot;
<font color="#EF2929">Error:</font> No such file or directory plan

Use the <font color="#34E2E2">--path</font> flag to specify the path to one of the following:
 - Terraform plan JSON file
 - Terraform directory
 - Terraform plan file
 - Terraform state JSON file
</pre>`
`<pre> make -s run ARGS=&quot;breakdown --path plan.json&quot;
<font color="#EF2929">Error:</font> Could not detect path type for plan.json

Use the <font color="#34E2E2">--path</font> flag to specify the path to one of the following:
 - Terraform plan JSON file
 - Terraform directory
 - Terraform plan file
 - Terraform state JSON file
</pre>`

### CODE REVIEW

File: cmd/infracost/run.go [hunk 1]
```diff
@@ -38,6 +38,18 @@ func runMain(cmd *cobra.Command, cfg *config.Config) error {
...
 		if provider == nil {
+			m := fmt.Sprintf("No such file or directory %s\n\n", ui.DisplayPath(projectCfg.Path))
+			m += fmt.Sprintf("Use the %s flag to specify the path to one of the following:\n", ui.PrimaryString("--path"))
+			m += " - Terraform plan JSON file\n - Terraform directory\n - Terraform plan file"
+
+			if cmd.Name() != "diff" {
+				m += "\n - Terraform state JSON file"
+			}
+
+			return errors.New(m)
```
Comment (aliscott) [inline cmd/infracost/run.go:49]:
```suggestion
			return events.NewError(errors.New(m), "No such file or directory ")
```
This sanitizes the path out of any error reported upstream, while still keeping the original error message to show the user.

**Review by aliscott — COMMENTED** [review]
I think we may need to update the `providers.Detect` function to return an error as well. I don't think this distinguishes between the file not existing and the path type not being detected.

Currently if `provider == nil` it can be either. So maybe we need the `providers.Detect` method to return different errors for `Could not detect path type` and `No such file or directory`.

File: cmd/infracost/run.go (hunk 1 above)
Comment (Pauluson) [reply cmd/infracost/run.go:49]: ah, okey, i understand

File: cmd/infracost/run.go [hunk 2]
```diff
@@ -35,10 +36,10 @@ func runMain(cmd *cobra.Command, cfg *config.Config) error {
 	projects := make([]*schema.Project, 0)
 
 	for _, projectCfg := range cfg.Projects {
-		provider := providers.Detect(cfg, projectCfg)
+		provider, err := providers.Detect(cfg, projectCfg)
 
 		if provider == nil {
```
Comment (aliscott) [inline cmd/infracost/run.go:41]:
```suggestion
		if err != nil {
```

**Review by aliscott — APPROVED** [review]
LGTM!

**Pull request merged**